From 2ed59daf9bdd345dd14a1598481410686c27f3e9 Mon Sep 17 00:00:00 2001
From: csagan5 <32685696+csagan5@users.noreply.github.com>
Date: Sat, 6 Nov 2021 09:46:24 +0100
Subject: Never use HTTP probes for connection detection

This patch disables the use of HTTP probes on Android < M devices
Connection will be always considered valid on such devices

License: GPL-3.0-only - https://spdx.org/licenses/GPL-3.0-only.html
---
 .../browser/net/connectivitydetector/ConnectivityDetector.java | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/chrome/android/java/src/org/chromium/chrome/browser/net/connectivitydetector/ConnectivityDetector.java b/chrome/android/java/src/org/chromium/chrome/browser/net/connectivitydetector/ConnectivityDetector.java
index 9cd998084f9c7..24fc75a362a03 100644
--- a/chrome/android/java/src/org/chromium/chrome/browser/net/connectivitydetector/ConnectivityDetector.java
+++ b/chrome/android/java/src/org/chromium/chrome/browser/net/connectivitydetector/ConnectivityDetector.java
@@ -182,7 +182,8 @@ public class ConnectivityDetector implements NetworkChangeNotifier.ConnectionTyp
 
         @Override
         public boolean shouldSkipHttpProbes() {
-            return false;
+            // always assume that connection is valid on Android < M
+            return true;
         }
     }
 
-- 
2.55.0

